src/error.ts

```diff
@@ -131,15 +135,28 @@ export class MongoError extends Error {
   cause?: Error; // depending on the node version, this may or may not exist on the base
 
   constructor(message: string | Error) {
+    super(MongoError.buildErrorMessage(message));
     if (message instanceof Error) {
-      super(message.message);
       this.cause = message;
-    } else {
-      super(message);
     }
+
     this[kErrorLabels] = new Set();
   }
 
+  /** @internal */
+  private static buildErrorMessage(e: Error | string): string {
```

Comment:
```suggestion
  private static buildErrorMessage(error: Error | string): string {
```

Reply:
I actually chose `e` intentionally, because it's not always an error (it can be a string).  Would you still like this changed?

Reply:
ah fair enough, I don't feel strongly